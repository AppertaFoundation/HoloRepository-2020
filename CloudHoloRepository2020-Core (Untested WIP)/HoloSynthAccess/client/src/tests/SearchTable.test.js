import React from 'react';
import { shallow } from 'enzyme';
import SearchTable from '../searchTable';
import renderer from 'react-test-renderer';
import { DropdownList } from 'react-widgets'

let wrapper = shallow(<SearchTable/>);

describe('Rendering', () => {
    it('Renders component correctly', () => {
    expect(wrapper).toMatchSnapshot();
    });

    it('should render one label', () => {
    expect(wrapper.find('label')).toHaveLength(1);
    });

    it('should render local form', () => {
        expect(wrapper.find('form')).toHaveLength(1);
    });

    it('should render local form search  text', () => {
        expect(wrapper.find('.searchLocalContainer').text()).toEqual('Search a body part locally:');
    });

    it('should render local inputs', () => {
        expect(wrapper.find('input')).toHaveLength(2);
    });


});

describe('Interactions', () => {
    it('should set local and body part', () => {
        wrapper.find('form').simulate('submit',{
         target: { value: "organ" },
         preventDefault: () => {
        }
    });
        expect(wrapper.state("bodyPart")).toEqual("organ");
        expect(wrapper.state.local).toEqual(false);
    });

    it('should set API and body part', () => {
        wrapper.find('.searchAPIContainer').simulate('change',
        {preventDefault: () => {
        }, target: { value: "organ" } 
    });
        expect(wrapper.state("bodyPart")).toEqual("organ");
        expect(wrapper.state.local).toEqual(true);
    });

    
    it('on change function is called', () => {
        const changeFn = jest.fn();
        const component = shallow(<DropdownList onChange={changeFn} />);
        component
        .find('Uncontrolled(DropdownList)')
        .simulate('change');
        expect(changeFn).toHaveBeenCalled();
    });
});