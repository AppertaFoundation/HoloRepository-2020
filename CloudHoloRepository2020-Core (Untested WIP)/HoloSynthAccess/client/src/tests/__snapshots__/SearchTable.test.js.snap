// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Renders component correctly 1`] = `
<div
  className="grid-container"
>
  <div
    className="searchLocalContainer"
  >
    <p>
      Search a body part locally:
    </p>
    <form
      onSubmit={[Function]}
    >
      <label>
        <input
          type="text"
        />
      </label>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
  <div
    className="searchAPIContainer"
  >
    <p>
      Select a body part:
    </p>
    <Uncontrolled(DropdownList)
      data={
        Array [
          "BRAIN",
          "CHEST",
          "BREAST",
          "PROSTATE",
          "COLON",
          "PHANTOM",
          "KIDNEY",
          "HEADNECK",
          "ABDOMEN",
          "NECK",
          "THORAX_1HEAD_NE",
          "Kidney",
          "OVARY",
          "LIVER",
          "LUNG",
          "BLADDER",
          "THYROID",
          "RECTUM",
          "UTERUS",
          "ESOPHAGUS",
          "STOMACH",
          "CERVIX",
          "CHESTABDPELVIS",
          "LEG",
          "TSPINE",
          "MEDIASTINUM",
          "SKULL",
          "EXTREMITY",
          "PANCREAS",
          "HEAD",
          "Phantom",
          "CT THORAX W CNT",
          "HEART",
          "CAROTID",
          "LUMBO-SACRAL SP",
          "PELVIS",
          "CHEST_TO_PELVIS",
          "CHEST_ABDOMEN",
          "J BRZUSZNA",
          "J brzuszna",
          "CT CHEST WO CE",
          "FUSION",
          "SEG",
          "TH CT CHEST WO ",
          "BD CT ABD WO_W ",
          "SELLA",
          "CHEST/ABD",
          "THORAX CT _OT01",
          "THORAX CT _AH05",
          "BRAIN W/WO_AH32",
          "OUTSIDE FIL",
          "CHEST COMPUTED ",
          "ABDOMEN_PELVIS ",
          "CHEST (THORAX) ",
          "CHEST NO GRID",
          "SPI CHEST 5MM",
          "AP PORTABLE CHE",
          "CTA CHEST",
          "CT CHEST W_ENHA",
          "Pelvis",
          "Head-and-Neck",
          "HEADANDNECK",
          "CT 3PHASE REN",
          "ABDOMENPELVIS",
          "ABD PELV",
          "ABD PEL",
          "WO INTER",
          "CAP",
          "Head-Neck",
          "PET_ABDOMEN_PEL",
          "PET_CT SCAN CHE",
          "THORAX",
          "WHOLEBODY",
        ]
      }
      onChange={[Function]}
    />
  </div>
  <div
    className="table"
  >
    <DataTable
      bodyPart=""
      local={false}
    />
  </div>
</div>
`;
